Apps/WebClient/src/ClientApp/src/services/interfaces.ts

```diff
@@ -91,10 +90,11 @@ export interface IUserProfileService {
     validateSMS(hdid: string, digit: string): Promise<boolean>;
     updateEmail(hdid: string, email: string): Promise<boolean>;
     updateSMSNumber(hdid: string, smsNumber: string): Promise<boolean>;
-    getUserPreference(hdid: string): Promise<UserPreference>;
-    createUserPreference(
-        userPreference: UserPreference
-    ): Promise<UserPreference>;
+    updateUserPreference(
+        hdid: string,
+        preference: string,
+        value: string
```

Comment:
We don't have to update yet, but we should always be pulling the version number.

Reply:
preferences are not represented as an object in the front end since they are a dictionary. with that in mind This update is also a create if you look at the delegate.